Openfeign integration not working when using hard-coded URLs

ghilainm opened the pull request:
Resolves #261

pavolloffay commented:
Could you please add a test and make sure CI passes?

ghilainm commented:
@pavolloffay I can't reproduce the issue in your project unfortunately. Seems like the Feign Client in my application is not found in the FeignContext which you rely on indirectly.

We then end up in 

`Targeter targeter = get(context, Targeter.class);
	return (T) targeter.target(this, builder, context,
			new HardCodedTarget<>(this.type, this.name, url));`

pavolloffay commented on instrument-starters/opentracing-spring-cloud-feign-starter/src/main/java/io/opentracing/contrib/spring/cloud/feign/TracingFeignBuilder.java:
```diff
@@ -0,0 +1,159 @@
+import feign.codec.Encoder;
+import feign.codec.ErrorDecoder;
+
+public class TracingFeignBuilder extends Feign.Builder {
```
Instead of wrapping this cannot we do something like this https://github.com/spring-cloud/spring-cloud-sleuth/blob/master/spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/feign/SleuthFeignBuilder.java

ghilainm replied:
We could but what if somebody has already put a builder in the context? In that case this code is ignored.

ghilainm replied:
I simplified implementation to not wrap Feign.Builder

pavolloffay commented:
Did you actually add test for your use case you are fixing - could you please point me to it? Or are you just refactoring the tests? If it is refactoring I would prefer to keep the original version to make sure changes are not affecting tests

ghilainm commented:
FeignClientWithoutClientInContextTest -> this is the test which is fixing the issue.

I have just changed one tests to isolate it from the others. I have also changed the global configuration to remove unnecessary configuration. I could restore it but I find it cleaner as you don't rely on it for the tests.

pavolloffay commented:
thanks @ghilainm
